test/Dialect/FIRRTL/imdce.mlir

```diff
@@ -453,3 +453,17 @@ firrtl.circuit "Test" {
     firrtl.strictconnect %out, %blah_out : !firrtl.uint<1>
   }
 }
+
+// -----
+// Test that a live use of a forceable declaration keeps it alive.
+// https://github.com/llvm/circt/issues/5898
+
+// CHECK-LABEL: circuit "Issue5898"
+firrtl.circuit "Issue5898" {
+  firrtl.module @Issue5898(in %x: !firrtl.uint<5>, out %p: !firrtl.rwprobe<uint<5>>) {
+    // CHECK: firrtl.wire
+    %w, %w_ref = firrtl.wire forceable : !firrtl.uint<5>, !firrtl.rwprobe<uint<5>>
+    firrtl.strictconnect %w, %x : !firrtl.uint<5>
+    firrtl.ref.define %p, %w_ref : !firrtl.rwprobe<uint<5>>
+  }
+}
```

Comment:
I don't think this test is actually checking anything. I tried running this on the firtool I had lying around and it preserves the wire.

Reply:
Oh, you're right.  It's the connect that was meant to be preserved, not sure what I was thinking.  Will fix and ensure  the test fails without the change (as with 5898).

Reply:
Fixed, and test fails without the change.  I'd checked the fir input (and my original code) worked with the change but oops the test.  Good catch, thank you!